common/selectors/selectAndDedupe.js

```diff
@@ -14,7 +16,21 @@ dedupe.defaults.createKey = site => site.cache_key || site.hostname || site.url;
 module.exports = function selectAndDedupe(group) {
   return group.reduce((result, options, index, arr) => {
     let current;
-    let sites = options.defaults ? options.sites.concat(options.defaults) : options.sites;
+    let sites = options.sites;
+
+    if (options.defaults) {
+      // Dedupe the defaults first.
+      let defaults = dedupe.group([sites, options.defaults])[1];
+      // Fill in any empty slots with default sites.
+      sites.forEach((site, i) => {
+        if (!site) {
+          sites[i] = defaults.shift();
```

Comment:
yes, this is possible. Maybe we show the empty placeholder tiles for that case? Otherwise it will show the pin for slot 12 at slot 8 or something like that.